### PR TITLE
Image Analysis SDK for Java

### PR DESCRIPTION
**Pull request by dargilco**
# Description

This is a new JAVA SDK for the Image Analysis service. It was created from [this TypeSpec PR](https://github.com/Azure/azure-rest-api-specs/pull/26146), followed by some API customization and adding samples, test code and README.md files.

# All SDK Contribution checklist:
- [v] **The pull request does not introduce [breaking changes]**
- [v] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [v] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [v] Title of the pull request is clear and informative.
- [v] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [v] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -160,6 +160,9 @@
 # PRLabel: %Open AI
 /sdk/openai/azure-ai-openai/                         @brandom-msft @jpalvarezl @mssfang
 
+# PRLabel: %Image Analysis
+/sdk/vision/azure-ai-vision-imageanalysis/           @dargilco @rhurey
```
Comment (jsquire) [inline .github/CODEOWNERS:164]: This label did not exist; I've created it with the proper configuration for our automation.

This line covers pull requests, but we'll also want to add an entry near the bottom of the file (looks like they start around L287) that defines who to contact for issues that we tag for service attention.   Assuming the same contacts, that would look like:

```text
# ServiceLabel: %Image Analysis %Service Attention
/sdk/vision/azure-ai-vision-imageanalysis/           @dargilco @rhurey
```

Comment (dargilco) [reply .github/CODEOWNERS:164]: Thank you @jsquire for adding the `Image Analysis` label, and the comment above! I will make this change in the next commit to this PR.

**Comment by haolingdong-msft** [comment]
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/ai/ImageAnalysis/tspconfig.yaml#L28
I saw `partial-update` is enabled for Java SDK. Could you point out all your partial updated code?

**Comment by dargilco** [comment]
@haolingdong-msft regarding your comment above, the file \custom\custom.md summarizes the manual changes I made to the auto-generated SDK code. Can you please review it and let me know the proper way of making these changes? Thanks!

File: sdk/vision/azure-ai-vision-imageanalysis/src/custom/custom.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+     *
+     * @return a string containing the bounding box values.
+     */
+    public String toString() {
+        return new StringBuilder().append("{x=").append(this.getX()).append(", y=").append(this.getY())
+            .append(", width=").append(this.getWidth()).append(", height=").append(this.getHeight()).append("}")
+            .toString();
+    }
+```
+
```
Comment (haolingdong-msft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/custom/custom.md:20]: Thanks for summarizing the partial-updated code! 
Happy to see the code are added to improve the user's usability(e.g. add overrides, or overwrite `toString()`), rather than adding workarounds.

File: sdk/vision/azure-ai-vision-imageanalysis/README.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+## Getting started
+
+### Prerequisites
+
+*  [Java Development Kit (JDK)](https://learn.microsoft.com/azure/developer/java/fundamentals/java-jdk-install) with version 8 or above.
+* An [Azure subscription](https://azure.microsoft.com/free).
+* A [Computer Vision resource](https://portal.azure.com/#create/Microsoft.CognitiveServicesComputerVision) in your Azure subscription.
+  * You will need the key and endpoint from this resource to authenticate against the service.
+  * You can use the free pricing tier (`F0`) to try the service, and upgrade later to a paid tier for production.
+  * Note that in order to run Image Analysis with the `Caption` or `Dense Captions` features, the Azure resource needs to be from one of the following GPU-supported regions: `East US`, `France Central`, `Korea Central`, `North Europe`, `Southeast Asia`, `West Europe`, or `West US`.
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/README.md:27]: Is there any documentation about this that we could link out to instead of writing it out explicitly here? I worry this may go stale over time.

And to tag along with the comment I left on the line above, could both these points be linked out to a getting starting type document for the service instead of being written out in the SDK README?

File: sdk/vision/azure-ai-vision-imageanalysis/README.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+```
+[//]: # ({x-version-update-end})
+
+### Set environment variables
+
+To authenticate the `ImageAnalysisClient`, you will need the endpoint and key from your Azure Computer Vision resource in the [Azure Portal](https://portal.azure.com). The code snippet below assumes these values are stored in environment variables:
+
+* Set the environment variable `VISION_ENDPOINT` to the endpoint URL. It has the form `https://your-resource-name.cognitiveservices.azure.com`, where `your-resource-name` is your unique Azure Computer Vision resource name.
+
+* Set the environment variable `VISION_KEY` to the key. The key is a 32-character Hexadecimal number.
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/README.md:47]: To be clear on these two points, are these environment configurations specific to the samples below or something the SDK will respect in a production environment?

File: sdk/vision/azure-ai-vision-imageanalysis/README.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+Once you define the environment variables, this Java code will create and authenticate a synchronous `ImageAnalysisClient`:
+
+```java imports-for-create-client-snippet
+import com.azure.ai.vision.imageanalysis.ImageAnalysisClient;
+import com.azure.ai.vision.imageanalysis.ImageAnalysisClientBuilder;
+import com.azure.core.credential.KeyCredential;
+```
+```java create-client-snippet
+String endpoint = System.getenv("VISION_ENDPOINT");
+String key = System.getenv("VISION_KEY");
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/README.md:60]: nit, for sample purposes I'd recommend using `azure-core`'s `Configuration` which can load configurations from many sources (defaulted to environment variables and Java system properties)

File: sdk/vision/azure-ai-vision-imageanalysis/README.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+
+```java imports-caption-file-snippet
+import com.azure.ai.vision.imageanalysis.ImageAnalysisOptionsBuilder;
+import com.azure.ai.vision.imageanalysis.models.ImageAnalysisResult;
+import com.azure.ai.vision.imageanalysis.models.VisualFeatures;
+import com.azure.core.util.BinaryData;
+import java.io.File;
+import java.util.Arrays;
+```
+```java caption-file-snippet
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/README.md:149]: I don't know if we want to include `import`s in our README code samples, I feel this can be left to the richer sample files.

@JonathanGiles I know you've done a lot of work this year around README documentation, is there any guidance around this?

File: sdk/vision/azure-ai-vision-imageanalysis/pom.xml [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      <artifactId>junit-jupiter-engine</artifactId>
+      <version>5.9.3</version> <!-- {x-version-update;org.junit.jupiter:junit-jupiter-engine;external_dependency} -->
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <version>4.11.0</version> <!-- {x-version-update;org.mockito:mockito-core;external_dependency} -->
+      <scope>test</scope>
+    </dependency>
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/pom.xml:75]: Something to handle supporting Java 21, add these additional dependencies:

```xml
<!-- bytebuddy dependencies are required for mockito 4.11.0 to work with Java 21. Mockito 4.11.0 is the last release -->
    <!-- of Mockito supporting Java 8 as a baseline. -->
    <dependency>
      <groupId>net.bytebuddy</groupId>
      <artifactId>byte-buddy</artifactId>
      <version>1.14.8</version> <!-- {x-version-update;testdep_net.bytebuddy:byte-buddy;external_dependency} -->
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>net.bytebuddy</groupId>
      <artifactId>byte-buddy-agent</artifactId>
      <version>1.14.8</version> <!-- {x-version-update;testdep_net.bytebuddy:byte-buddy-agent;external_dependency} -->
      <scope>test</scope>
    </dependency>
```

File: sdk/vision/azure-ai-vision-imageanalysis/src/main/java/com/azure/ai/vision/imageanalysis/ImageAnalysisOptionsBuilder.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.ai.vision.imageanalysis;
+
+import java.util.List;
+
+/**
+ *  Builder class to set options for a single Image Analysis operation.
+ */
+public final class ImageAnalysisOptionsBuilder {
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/main/java/com/azure/ai/vision/imageanalysis/ImageAnalysisOptionsBuilder.java:10]: Was there a specific reason to have the Options type have a builder? Generally, Options types are fluent and not immutable.

File: sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/run_all_samples.cmd [hunk 9]
```diff
@@ -0,0 +1,39 @@
+REM
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/run_all_samples.cmd:1]: nit: we should opt for PowerShell scripts given their ability to run on all major OSes we support

File: sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/pom.xml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>2.0.7</version>
+    </dependency>
+    <!-- https://mvnrepository.com/artifact/org.slf4j/slf4j-simple -->
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <version>2.0.7</version>
+    </dependency>
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/pom.xml:26]: This POM could be removed and these dependencies added as test dependencies in the SDK POM. Samples are compiled during test compilation.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java [hunk 11]
```diff
@@ -0,0 +1,605 @@
...
+        assertEquals(polygon.size(), 4);
+        for (int i = 0; i < polygon.size(); i++) {
+            assertTrue(polygon.get(i).getX() > 0);
+            assertTrue(polygon.get(i).getY() > 0);
+        }
+    }
+
+    private static void printAnalysisResults(String testName, ImageAnalysisResult result) {
+
+        System.out.println(" ******************** TEST NAME: " + testName + " ******************** ");
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:534]: If it isn't required, I'd only run this if a test failed. Printing on all tests can make it harder to parse output from test runs to find the cause on failing tests (stack traces from failed tests, etc)

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisAsyncClientTest.java [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+        visualFeatures =  Arrays.asList(VisualFeatures.SMART_CROPS);
+        options = new ImageAnalysisOptionsBuilder().setSmartCropsAspectRatios(Arrays.asList(0.9, 1.33)).build();
+        doAnalysis(methodName + ":SmartCrops", sync, imageSource, visualFeatures, options);
+
+        visualFeatures =  Arrays.asList(VisualFeatures.TAGS);
+        options = new ImageAnalysisOptionsBuilder().setLanguage("en").build();
+        doAnalysis(methodName + ":Tags", sync, imageSource, visualFeatures, options);
+
+        visualFeatures =  Arrays.asList(VisualFeatures.PEOPLE);
+        doAnalysis(methodName + ":People", sync, imageSource, visualFeatures, null);
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisAsyncClientTest.java:65]: I'd recommend turning this into multiple tests, if the first one fails we don't know if the latter ones fail and may need to iterate many times before the test passes.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTest.java [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+        ImageAnalysisOptions options = null;
+
+        List<VisualFeatures> visualFeatures =  Arrays.asList(VisualFeatures.CAPTION);
+        doAnalysis(methodName + ":Caption", sync, imageSource, visualFeatures, options);
+
+        visualFeatures =  Arrays.asList(VisualFeatures.READ);
+        doAnalysis(methodName + ":Read", sync, imageSource, visualFeatures, options);
+
+        visualFeatures =  Arrays.asList(VisualFeatures.TAGS);
+        doAnalysis(methodName + ":Tags", sync, imageSource, visualFeatures, options);
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTest.java:76]: Same comment with multiple tests

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java [hunk 14]
```diff
@@ -0,0 +1,605 @@
...
+            validateRead(result);
+        } else {
+            assertNull(result.getRead());
+        }
+    }
+
+    private static void validateMetadata(ImageAnalysisResult result) {
+        assertNotNull(result.getMetadata());
+        assertEquals(result.getMetadata().getHeight(), 576);
+        assertEquals(result.getMetadata().getWidth(), 864);
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:269]: nit: first value is the expected value, second value is the actual value. Tests will still run correctly flipped but the error logged by JUnit will be backwards.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java [hunk 15]
```diff
@@ -0,0 +1,605 @@
...
+    }
+
+    private static void validateModelVersion(ImageAnalysisResult result) {
+        assertNotNull(result.getModelVersion());
+        assertFalse(result.getModelVersion().isEmpty());
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:274]: I think this should be removed, if it ends up being empty this error message produced by `assertFalse` will be more confusing than `assertEquals` telling you `excpected: 2023-10-01, got: ""`

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java [hunk 16]
```diff
@@ -0,0 +1,605 @@
...
+        assertTrue(found1);
+        assertTrue(found2);
+
+        // Make sure each tag is unique
+        for (int i = 0; i < tagsResult.getValues().size(); i++) {
+            for (int j = i + 1; j < tagsResult.getValues().size(); j++) {
+                assertFalse(
+                    tagsResult.getValues().get(j).getName().equals(
+                    tagsResult.getValues().get(i).getName()));
+            }
```
Comment (alzimmermsft) [inline sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:413]: Does this test make sense? This is comparing object types without an `equals` overload, which unless they are pointing to the same object will always be false.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java (hunk 15 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:274]: Thanks @alzimmermsft. Agree! the line `assertFalse(result.getModelVersion().isEmpty());` will be removed in the next commit.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java (hunk 14 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:269]: thanks @alzimmermsft ! The next commit will include a change to make sure expected values come first.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java (hunk 16 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:413]: @alzimmermsft The `.getName()` method here returns a regular Java `String` object. I need to make sure that every tag name (a String) in the list of returned image tags is unique. So I check all pairs and compare two Java strings, with the built-in String.equals() method. I double checked and the code should be good as is. Please reach out to me over IM if you still have comments about this. Thank you!

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisAsyncClientTest.java (hunk 12 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisAsyncClientTest.java:65]: @alzimmermsft thank you for this comment. Some tests do a single Image Analysis call, other tests (like the ones you pointed out) do multiple Image Analysis calls using the same ImageAnalysisClient. It's important that we have a test or two that do multiple calls using the same client, as this will be the common use case (creating a single client and using it to analyze, for example, a folder full of images). So, I'd rather keep this test as-is if that's okay with you. Reach out to me over IM if you want to discuss this further.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTest.java (hunk 13 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTest.java:76]: Thanks! see my comment in ImageAnalysisAsyncClientTest.java

File: sdk/vision/azure-ai-vision-imageanalysis/README.md (hunk 6 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/README.md:149]: @alzimmermsft @JonathanGiles I see this README.md is a self-contained quick-start guide. My expectation is that developers can copy this code and run it quickly. I don't want them to try and figure out which packages each class comes from. Unless this violates some hard guideline, I'd like to keep it as is. Thanks!

File: sdk/vision/azure-ai-vision-imageanalysis/README.md (hunk 5 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/README.md:60]: Hi @alzimmermsft, thank you for this comment! I looked at the Configuration class (I was not familiar with it). I tried it out. Looks like instead of 
```java
String endpoint = System.getenv("VISION_ENDPOINT");
String key = System.getenv("VISION_KEY");
```
I can use:
```java
import  com.azure.core.util.Configuration;
import  com.azure.core.util.ConfigurationBuilder;

Configuration configuration = new ConfigurationBuilder().build();
String endpoint = configuration.get("VISION_ENDPOINT");
String key = configuration.get("VISION_KEY");
```
That worked great. However, I want to keep the samples simple and not introduce any new dependencies that will require developers to learn a new class unrelated to the Image Analysis SDK. I see that using System.getenv() is very common in sample code, so I'd rather keep it as is if that's okay with you.

File: sdk/vision/azure-ai-vision-imageanalysis/README.md (hunk 4 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/README.md:47]: @alzimmermsft good point, I added this sentence to make it clear: 

_Note that the client library does not directly read these environment variable at run time. The endpoint and key must be provided to the `ImageAnalysisClientBuilder` in your code. The code snippet below reads environment variables to promote the practice of not hard-coding secrets in your source code._

File: sdk/vision/azure-ai-vision-imageanalysis/README.md [hunk 17]
```diff
@@ -0,0 +1,340 @@
...
+
+## Getting started
+
+### Prerequisites
+
+*  [Java Development Kit (JDK)](https://learn.microsoft.com/azure/developer/java/fundamentals/java-jdk-install) with version 8 or above.
+* An [Azure subscription](https://azure.microsoft.com/free).
+* A [Computer Vision resource](https://portal.azure.com/#create/Microsoft.CognitiveServicesComputerVision) in your Azure subscription.
+  * You will need the key and endpoint from this resource to authenticate against the service.
+  * You can use the free pricing tier (`F0`) to try the service, and upgrade later to a paid tier for production.
```
[the remark this answers is not in the document]
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/README.md:26]: @alzimmermsft Removed. Thanks!

File: sdk/vision/azure-ai-vision-imageanalysis/README.md (hunk 3 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/README.md:27]: @alzimmermsft Instead of listing them here, I updated the text and I added a link to public learn.microsoft page listing the supported GPU region. Thanks!

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java (hunk 11 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:534]: @alzimmermsft I made a change to (by default) not print all the results. This is controlled by a final boolean variable in `ImageAnalysisClientTestBase.java`. So you can enable it in local runs if needed.

File: sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/run_all_samples.cmd (hunk 9 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/run_all_samples.cmd:1]: @alzimmermsft I added equivalent PowerShell scripts. Thanks!

File: sdk/vision/azure-ai-vision-imageanalysis/README.md (hunk 5 above)
Comment (alzimmermsft) [reply sdk/vision/azure-ai-vision-imageanalysis/README.md:61]: I see no issue with leaving the sample using System.getenv(). `Configuration` should automatically be available as a transitive dependency from `azure-core`, but it does add some cognitive overhead as it's specific to the Azure SDKs and not a Java language concept.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java (hunk 16 above)
Comment (alzimmermsft) [reply sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java:418]: Oh sorry, missed the `getName` call, this makes more sense now

File: sdk/vision/azure-ai-vision-imageanalysis/pom.xml (hunk 7 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/pom.xml:75]: Thanks @alzimmermsft, I added these two dependencies to the root pom.xml.

File: sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/pom.xml (hunk 10 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/samples/java/com/azure/ai/vision/imageanalysis/pom.xml:26]: @alzimmermsft I would like to offer developers the option to build the samples and test them against the public package. Therefore I have this separate pom.xml next to the samples source code. The samples README.md shows how to build and run all or a single sample.
Of course I would also like the ADO pipeline to automatically build the samples after building the SDK and running tests. I'll look at the pipline to make sure samples are built. The room pom.xml already has >slf4j-simpl (which is required during compile time). slf4j-api package is required during run-time if you turn on logging. Some verification tool does not allow me to add this dependency in the root pom.xml. I'm assuming for a good reason. 
So bottom line: I prefer keeping things as is, while validating that indeed samples are compiled as part of ADO pipeline. I'll keep this comment open until I verify.

File: sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java (hunk 16 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java]: @alzimmermsft good point, I switched to using assertNotEquals. Thanks!

File: sdk/vision/azure-ai-vision-imageanalysis/src/main/java/com/azure/ai/vision/imageanalysis/ImageAnalysisOptionsBuilder.java (hunk 8 above)
Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/main/java/com/azure/ai/vision/imageanalysis/ImageAnalysisOptionsBuilder.java:10]: After further discussion we agreed to remove the ImageAnalysisOptionsBuilder class. I will submit a commit to this PR that will update ImageAnalysisOptions with new "setXxx" methods that support chained calls. I will leave this comment open until the commit is done.

Comment (dargilco) [reply sdk/vision/azure-ai-vision-imageanalysis/src/main/java/com/azure/ai/vision/imageanalysis/ImageAnalysisOptionsBuilder.java]: Resolving after latest commit that removed ImageAnalysisOptionsBuilder. Thanks @alzimmermsft!
